### PR TITLE
validate params to internalRGB

### LINKED ISSUE #54: throw error if bad values passed to internalRGB
`if (red === null || red === undefined) { throw ... }`

### PR DESCRIPTION
**Pull request by Resseguie**
Fixes #54

### CODE REVIEW

File: lib/spark.js [hunk 1]
```diff
@@ -440,6 +440,10 @@ Spark.prototype.internalRGB = function(red, green, blue) {
     input = red;
 
     if (typeof input === "object") {
+      if (input === null || input === undefined) {
+        throw new Error("Spark.internalRGB: invalid color (" + input + ")");
+      }
```
Comment (rwaldron) [inline lib/spark.js]: This should come before `if (typeof input === "object") {`

File: lib/spark.js [hunk 2]
```diff
@@ -461,6 +465,10 @@ Spark.prototype.internalRGB = function(red, green, blue) {
           input = input.slice(1);
         }
 
+        if (input.length !== 6) {
```
Comment (rwaldron) [inline lib/spark.js]: Hadn't even considered this, good catch

**Comment by rwaldron** [comment]
Needs tests

```js
test.throws(function() {
   ...
});
```

**Comment by Resseguie** [comment]
@rwaldron my first tests with test.throws. Care to sanity check?

File: lib/spark.js [hunk 3]
```diff
@@ -461,6 +466,10 @@ Spark.prototype.internalRGB = function(red, green, blue) {
           input = input.slice(1);
         }
 
+        if (!input.match(/[0-9A-Fa-f]{6}/)) {
```
Comment (rwaldron) [inline lib/spark.js]: Solid.

File: lib/spark.js [hunk 4]
```diff
@@ -470,6 +479,15 @@ Spark.prototype.internalRGB = function(red, green, blue) {
       }
     }
   } else {
+    
+    if (
+      red === null || red === undefined ||
+      green === null || green === undefined ||
+      blue === null || blue === undefined
```
Comment (rwaldron) [inline lib/spark.js]: Feel free to change `eqnull: true,` to `eqnull: false,` in Gruntfile.js, that will simplify this

File: test/spark.js [hunk 5]
```diff
@@ -877,5 +877,35 @@ exports["Spark.prototype.internalRGB"] = {
...
+      this.spark.internalRGB(10, 20, null);
+    });
+
+    test.throws(function() {
+      this.spark.internalRGB([10, 20, null]);
+    });
+
+    test.throws(function() {
+      this.spark.internalRGB({red: 255, green: 100});
+    });
```
Comment (rwaldron) [inline test/spark.js]: Add one that includes `undefined`, like: 

```js
test.throws(function() {
  this.spark.internalRGB(null, undefined, 255);
});
```

Comment (Resseguie) [inline test/spark.js]: Good catch. I'll add undefined check for params, array, and object setting.

File: lib/spark.js (hunk 4 above)
Comment (Resseguie) [inline lib/spark.js]: I moved these checks down in my later commit so that at least I didn't have to repeat them for the array, object, three param, etc.

**Comment by Resseguie** [comment]
@rwaldron sorry for all the extra commits instead of --amend. Network I'm on has funny git alias set up and long story short, I'm limited to what the git UI offers me...

File: lib/spark.js (hunk 3 above)
Comment (Resseguie) [inline lib/spark.js]: thanks, except it should have included ^ and $ to match whole string. Fixed in updated commit below, along with extra tests to provide more regex coverage...

File: lib/spark.js [hunk 6]
```diff
@@ -474,6 +481,16 @@ Spark.prototype.internalRGB = function(red, green, blue) {
     values = [red, green, blue];
   }
 
+  // Be sure we got all three values from one of the above signatures
+  if (
+    values.length !== 3 ||
+    values[0] === null || values[0] === undefined ||
+    values[2] === null || values[1] === undefined ||
+    values[2] === null || values[2] === undefined
```
Comment (rwaldron) [inline lib/spark.js]: Can still relax jshint with eqnull: false and simplify this

Comment (Resseguie) [inline lib/spark.js]: I must have had blinders on earlier. It just clicked what you were saying. Got it. I'll update both repos.

Comment (Resseguie) [inline lib/spark.js]: @rwaldron isn't it jshint eqnull:true that actually _allows_ the x == null check? In that case, it's already set as so in the Gruntfile and I can already add the simplification.

And wow, glad I'm looking at this again. Check out the aweful cut-n-paste error I had there at line 488. checking values[2] for null twice... exactly why simplifying this is a good idea.

**Comment by rwaldron** [comment]
LGTM

**Pull request merged**